docs/source/overview.rst

```diff
@@ -63,7 +63,47 @@ Main concepts
 Deploy
 ^^^^^^
 
-will be added soon
+
+Deployment is designed in the way, that user pays for himself. To do that, we use counterfactual deployment. (``create2`` function to deploy contract)
+
+The process looks like follows:
+
+- **computing contract address** - SDK computes deterministic contract address. Contract address is unique and is signed to the users public key.
+
+- **waiting for balance** - User needs to send funds on this address. He can transfer ether on his own or use on-ramp provider. SDK wait for future contract address balance change. If SDK discovers, that required funds are on this balance, sends DEPLOY request to relayer.
+
+- **deploy** - Relayer deploys the contract and immediately gets the refund from the contract.
```

Comment:
- **refund** during deployment contract will refund the cost of the transaction to relayer address